apisix/plugins/opa/helper.lua

```diff
@@ -45,16 +48,68 @@ local function build_http_request(conf, ctx)
 end
 
 
-function _M.build_opa_input(conf, ctx, subsystem)
-    local request = build_http_request(conf, ctx)
+local function build_http_route(conf, ctx, remove_upstream)
+    local route = core.table.clone(ctx.matched_route).value
+
+    if remove_upstream and route and route.upstream then
```

Comment:
what is `remove_upstream` used? I see that true is passed directly.

Reply:
Yes, it has been used to generate upstream data before, when it did not need to be removed. It is possible that it will be used again in the future, should I remove it now?

Reply:
ok, keep it.